src/gt4py/next/type_system/type_info.py

```diff
@@ -754,3 +754,18 @@ def accepts_args(
         return True
 
     return next(errors, None) is None
+
+
+def flatten(arg: list | tuple | ts.TupleType):
```

Comment:
This will be in another PR, but I think the most elegant solution would be to make `TupleType` iterable and then simply use `eve.utils.flatten_iter`. This might actually be enough for all the other implementations too.

For reference: `primitive_constituents` doesn't work on sequences.

Reply:
> was this moved? maybe add TODO with link to #1395

I commented on #1395 with the approach taken in #1487.

Reply:
:+1: